### ISSUE TITLE
Issue with Date polyfil and IE9

### ISSUE THREAD

**Issue by GarthMartin** [wet-boew/wet-boew#6553]
Strange issue with IE9 and the date polyfil.  For some reason you have to click twice on the icon to get the calendar to open?  Code below:  

Also, is there any way to have the calendar open when the input element recieved focus? (Nice to have of course - is there a reason it does not?)
Using 4.0.10

Thanks in advance,

Garth
```
<div>
    <label for="RequestedDateValue">Requested Date
        <span class="datepicker-format">(<abbr title="Four digits year, dash, two digits month, dash, two digits day">YYYY-MM-DD</abbr>)</span>
    </label>
    <input id="RequestedDateValue" type="date" name="RequestedDateValue"/>
    </div>
```

**Comment by GarthMartin** [comment]
Sorry just to expand a touch, when you click the first time, the icon moves below the label and then back up beside the input element.  The second time, the date picker opens.  I put an alert into the 
```javascript
$("input[type=date]").on("wb-ready.wb-date", function (event)
```
event that fires on load, so as far as I can tell the date picker is inited, but it does not load.  Note the alert does not fire when you click the icon the first time (nor the second time), and after the first click, any and all other date pickers on the page (none in example) open on single click.

Cheers,

Garth

**Comment by LaurentGoderre** [comment]
Wonder if it's related to #6545

**Comment by GarthMartin** [comment]
#6545 seems to be a reversed scenario where it initially works, but after that it takes 2 clicks.

**Comment by pjackson28** [comment]
Note that the calendar can't be opened when the input receives focus because that doesn't happen when the browser has native support for input type="date" (you have to click the button to open the date picker). We are just providing a polyfill to add support to browsers that lack the native support.

**Comment by neoinsight** [comment]
Also don't want the calendar to open automatically because many people prefer to just type in the date, and are more successful and faster at doing that than using the calendar.

**Comment by LaurentGoderre** [comment]
:+1:

**Comment by LaurentGoderre** [comment]
I can't reproduce (see SauceLabs test)

https://saucelabs.com/tests/13c9ea6a04c54642850c6fbb24db3fb3

**Comment by bellisrartech** [comment]
I have the exact same issue as GarthMartin in FireFox (currently v36) and IE8 and IE10. I get a gray line with the calendar icon shifting to the left the very first time that I click on the calendar glyph after the page has loaded initializing. After that the calendar control displays properly. Here is my markup:
<div class="form-group">
 <label class="col-md-3 control-label" for="date1"><span class="field-name">from</span><span class="datepicker-format"> (YYYY-MM-DD)</span></label>
 <input class="col-md-5 form-control input-group" id="date1" name="date1" type="date" data-rule-dateISO="true"/>
</div>

This was originally taken from the http://wet-boew.github.io/wet-boew/demos/formvalid/formvalid-en.html working example.

This control does work properly from the working example, so I'm not sure why I'm seeing this behavior in my page.

**Comment by GarthMartin** [comment]
@LaurentGoderre : I cannot access the saucelabs page,but I have narrowed this down to the following issue.  If I wrap my main ContentPlaceHolder (in master page) in the following tag, I no longer get the double click behaviour.

``` html
<main role="main" property="mainContentOfPage">
                <asp:ContentPlaceHolder ID="MainContent" runat="server">
                </asp:ContentPlaceHolder>
</main>
```
That being said, if I have the datepicker inside a lightbox, I get an Error: DOM Exception: HIERARCHY_REQUEST_ERR (3) if I submit the form having used the datepicker, but not if I enter a date manually.  This may or may not be related to an error with the lightbox when it loads:

``` javascript
Modernizr.load({
			load: "site!deps/jquery.magnific-popup" + wb.getMode() + ".js",
			complete: function() {

				// Set the dependency i18nText only once
				$.extend( true, $.magnificPopup.defaults, i18nText );
				extendedGlobal = true;

				$document.trigger( dependenciesLoadedEvent );
			}
		});

```
indicating that it cannot find the property "defaults" of  $.magnificPopup.

I have not investigated this in depth yet. 

What is the purpose of the <main> tag other than for separation of content?  Why would it effect the way the polyfill loads?

**Comment by LaurentGoderre** [comment]
It isw possible that .NET injects some JKavascript that conflicts with WET....it's pretty bad for that.

What tag are you referring to?

**Comment by GarthMartin** [comment]
@LaurentGoderre Sorry that was perhaps not clear...I wrapped the ContentPlaceHolder in the <main> tag with the role and property as you see in the previous comment.

**Comment by LaurentGoderre** [comment]
@GarthMartin it seems probable to me that .NET is injecting some JavaScript that interferes with it.

**Comment by hsrudnicki** [comment]
@GarthMartin @LaurentGoderre and others: Can this issue be closed or otherwise?

**Comment by bellisrartech** [comment]
As far as I'm concerned it can be closed.

Regards,
Bill Ellis
Raritan Technologies, Inc.
Phone:908-668-8181 x112
E-Mail: wsellis@raritantechnologies.com

**Comment by LaurentGoderre** [comment]
@GarthMartin how about you?
